### ISSUE TITLE
[macOS/Windows] Accessibility bridge should remove its tree manager

### ISSUE THREAD

**Issue by loic-sharma** [flutter/flutter#188662]
### Problem

The `AccessibilityBridge` adds its `AXTreeManager` here:

https://github.com/flutter/flutter/blob/bbb7a56bdce2c4d96dbe2d519f1077fff15025c3/engine/src/flutter/shell/platform/common/accessibility_bridge.cc#L23-L32

However, it does not remove it here:

https://github.com/flutter/flutter/blob/bbb7a56bdce2c4d96dbe2d519f1077fff15025c3/engine/src/flutter/shell/platform/common/accessibility_bridge.cc#L34-L37

### Suggested fix

Add the missing cleanup call to the top of the destructor:

```cpp
AccessibilityBridge::~AccessibilityBridge() {
  ui::AXTreeManagerMap::GetInstance().RemoveTreeManager(tree_->GetAXTreeID());
  // ... existing cleanup
}
```

### Internal bug

[b/525534446](https://buganizer.corp.google.com/issues/525534446)
